openmc/data/decay.py

```diff
@@ -0,0 +1,458 @@
+from collections import Iterable, namedtuple
+from io import StringIO
+from math import log
+from numbers import Real
+import re
+from warnings import warn
+
+from six import string_types
+import numpy as np
+try:
+    from uncertainties import ufloat, unumpy, UFloat
+except ImportError:
+    ufloat = UFloat = namedtuple('UFloat', ['nominal_value', 'std_dev'])
+
+import openmc.checkvalue as cv
+from openmc.mixin import EqualityMixin
+from .data import ATOMIC_SYMBOL, ATOMIC_NUMBER
+from .endf import Evaluation, get_head_record, get_list_record, get_tab1_record
+
+
+# Gives name and (change in A, change in Z) resulting from decay
+_DECAY_MODES = {
+    0: ('gamma', (0, 0)),
+    1: ('beta-', (0, 1)),
+    2: ('ec/beta+', (0, -1)),
+    3: ('IT', (0, 0)),
+    4: ('alpha', (-4, -2)),
+    5: ('n', (-1, 0)),
+    6: ('sf', None),
+    7: ('p', (-1, -1)),
+    8: ('e-', (0, 0)),
+    9: ('xray', (0, 0)),
+    10: ('unknown', None)
+}
+
+_RADIATION_TYPES = {
+    0: 'gamma',
+    1: 'beta-',
+    2: 'ec/beta+',
+    4: 'alpha',
+    5: 'n',
+    6: 'sf',
+    7: 'p',
+    8: 'e-',
+    9: 'xray',
+    10: 'anti-neutrino',
+    11: 'neutrino'
+}
+
+
+def get_decay_modes(value):
+    """Return sequence of decay modes given an ENDF RTYP value.
+
+    Parameters
+    ----------
+    value : float
+        ENDF definition of sequence of decay modes
+
+    Returns
+    -------
+    list of str
+        List of successive decays, e.g. ('beta-', 'neutron')
+
+    """
+    return [_DECAY_MODES[int(x)][0] for x in
+            str(value).strip('0').replace('.', '')]
+
+
+class FissionProductYields(EqualityMixin):
+    """Independent and cumulative fission product yields.
+
+    Parameters
+    ----------
+    ev_or_filename : str of openmc.data.endf.Evaluation
+        ENDF fission product yield evaluation to read from. If given as a
+        string, it is assumed to be the filename for the ENDF file.
+
+    Attributes
+    ----------
+    cumulative : list of dict
+        Cumulative yields for each tabulated energy. Each item in the list is a
+        dictionary whose keys are nuclide names and values are cumulative
+        yields. The i-th dictionary corresponds to the i-th energy.
+    energies : Iterable of float or None
+        Energies at which fission product yields are tabulated.
```

Comment:
If we were to do this, then I would need to make some sort of ad hoc scheme for determining what spectrum is used given the energy that the evaluator specified. ENDF/B-VII.1 and JEFF 3.1.1 don't use the same energies, for example. They agree on 0.0253 eV for thermal, but ENDF has yields at 500 keV whereas JEFF uses 400 keV. Pu239 in ENDF/B-VII.1 actually has four sets of yields (an additional one at 2 MeV). So, what I'll do for now is leave the energies as is but add a `Notes` section to the documentation saying that they shouldn't be taken literally.